### ISSUE TITLE
App does not show incoming pending Ethereum trancastions

### ISSUE THREAD

**Issue by Tomasvrba** [digitalbitbox/bitbox-wallet-app#837]
Only outgoing transactions are saved to the db and showed as pending, incoming transactions are not shown in the app until they have at least one confirmation. This will probably require Infura (or access to a node) since afaik, etherscan doesn't support the `eth_pendingTransactions` json rpc.

**Comment by benma** [comment]
`eth_pendingTransactions`'s description is a bit unclear - it seems to return all pending txs of everyone, unfiltered? That sounds very inefficient.

Infura or other normal nodes unfortunately do not have a transaction index :(

**Comment by Tomasvrba** [comment]
> `eth_pendingTransactions`'s description is a bit unclear - it seems to return all pending txs of everyone, unfiltered? That sounds very inefficient.

Yeah that's the case unfortunately afaik, you would have to get the pool of all pending transactions from a node (looks like even Infura doesn't offer this rpc) and then filter out the ones you want. Not a very urgent feature I suppose :D
